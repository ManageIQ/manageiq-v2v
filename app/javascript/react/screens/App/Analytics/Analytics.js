import React from 'react';
import PropTypes from 'prop-types';
import { Breadcrumb, EmptyState, Button } from 'patternfly-react';
import Toolbar from '../../../config/Toolbar';
import BreadcrumbPageSwitcher from '../common/BreadcrumbPageSwitcher';

const AnalyticsContainer = ({ showEmptyState, children }) =>
  showEmptyState ? (
    <div className="row cards-pf" style={{ overflow: 'auto', overflowX: 'hidden', paddingBottom: 50, height: '100%' }}>
      {children}
    </div>
  ) : (
    children
  );

AnalyticsContainer.propTypes = {
  showEmptyState: PropTypes.bool,
  children: PropTypes.node
};

const Analytics = ({ showEmptyState = true }) => (
  <React.Fragment>
    <Toolbar>
      <Breadcrumb.Item active>{__('Compute')}</Breadcrumb.Item>
      <Breadcrumb.Item active>{__('Migration')}</Breadcrumb.Item>
      <Breadcrumb.Item active>
        <strong>{__('Migration Analytics')}</strong>
      </Breadcrumb.Item>
      <BreadcrumbPageSwitcher activeHref="#/analytics" />
    </Toolbar>
    <AnalyticsContainer showEmptyState={showEmptyState}>
      {showEmptyState && (
        <EmptyState className="full-page-empty">
          <EmptyState.Icon type="pf" name="search" />
          <EmptyState.Title>
            {__('Examine your virtual environment using Red Hat Migration Analytics')}
          </EmptyState.Title>
          <EmptyState.Info>
            {__('Get a summarized report of your virtualization providers in CloudForms and collect a detailed inventory that can be further analyzed using the Migration Analytics application on cloud.redhat.com.') /* prettier-ignore */}
          </EmptyState.Info>
          <EmptyState.Action>
            <Button bsStyle="primary" bsSize="large" onClick={() => alert('TODO: modal here')}>
              {__('Get started')}
            </Button>
          </EmptyState.Action>
        </EmptyState>
      )}
      <h6 className="manifest-version">{__('Manifest version: x.y.z 2019-06-10') /* TODO make this real */}</h6>
    </AnalyticsContainer>
  </React.Fragment>
);

Analytics.propTypes = {
  showEmptyState: PropTypes.bool
};

export default Analytics;
