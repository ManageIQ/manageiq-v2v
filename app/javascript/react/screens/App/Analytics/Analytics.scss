.manifest-version {
  text-align: right;
  padding: 35px;
}
