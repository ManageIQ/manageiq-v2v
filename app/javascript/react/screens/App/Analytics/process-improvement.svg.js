import React from 'react';

export default props => (
  <svg data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 46.1 55.3" {...props}>
    <title>Icon_RH_Diagram_ProcessImprovement-B_RGB_Black</title>
    <path d="M19.28 8.03H39.2a4.9 4.9 0 1 1 0 9.79h-.65v-2l-5.18 3 5.18 3v-2h.7a6.9 6.9 0 1 0 0-13.79H19.28a1 1 0 1 0 0 2zm12.27 39.3a1 1 0 0 0 .71.29h.17a1 1 0 0 0 .73-.55l5.15-10.58a1 1 0 1 0-1.8-.87l-4.54 9.32-2.54-2.54a1 1 0 0 0-1.41 1.41zm13.36-18.52H20.42a1 1 0 0 0-1 1V54.3a1 1 0 0 0 1 1h24.49a1 1 0 0 0 1-1V29.81a1 1 0 0 0-1-1zm-1 24.49H21.42V30.81h22.49zM10.55 29.62h-.7a4.9 4.9 0 0 1 0-9.8h19.88a1 1 0 0 0 0-2H9.8a6.9 6.9 0 0 0 0 13.8h.75v2l5.18-3-5.18-3zm3.56-16.56V1a1 1 0 0 0-1-1H1a1 1 0 0 0-1 1v12.06a1 1 0 0 0 1 1h12.06a1 1 0 0 0 1-1zm-2-1H2V2h10.06z" />
  </svg>
);
