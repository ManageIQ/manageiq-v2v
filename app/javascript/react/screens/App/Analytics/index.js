import { connect } from 'react-redux';
import Analytics from './Analytics';

export const reducers = {}; // TODO add analytics reducer

const mapStateToProps = () => ({}); // TODO pull analytics state from store

const mergeProps = (stateProps, dispatchProps, ownProps) => Object.assign(stateProps, ownProps.data, dispatchProps);

export default connect(
  mapStateToProps,
  {}, // TODO add analytics actions
  mergeProps
)(Analytics);
