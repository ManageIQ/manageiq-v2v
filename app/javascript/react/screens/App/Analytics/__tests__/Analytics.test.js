import React from 'react';
import { shallow } from 'enzyme';
import Analytics from '../Analytics';

describe('Analytics component', () => {
  it('renders correctly', () => {
    const component = shallow(<Analytics />);
    expect(component).toMatchSnapshot();
  });
});
