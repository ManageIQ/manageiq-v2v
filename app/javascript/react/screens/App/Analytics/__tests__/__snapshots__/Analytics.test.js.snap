// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics component renders correctly 1`] = `
<React.Fragment>
  <Toolbar>
    <BreadcrumbItem
      active={true}
    >
      Compute
    </BreadcrumbItem>
    <BreadcrumbItem
      active={true}
    >
      Migration
    </BreadcrumbItem>
    <BreadcrumbItem
      active={true}
    >
      <strong>
        Migration Analytics
      </strong>
    </BreadcrumbItem>
    <BreadcrumbPageSwitcher
      activeHref="#/analytics"
    />
  </Toolbar>
  <AnalyticsContainer
    showEmptyState={true}
  >
    <EmptyState
      className="full-page-empty"
    >
      <div
        className="blank-slate-pf-icon"
        style={
          Object {
            "opacity": 0.4,
          }
        }
      >
        <Component
          height="80px"
        />
      </div>
      <EmptyStateTitle
        className=""
      >
        Examine your virtual environment using Red Hat Migration Analytics
      </EmptyStateTitle>
      <EmptyStateInfo
        className=""
      >
        Get a summarized report of your virtualization providers in CloudForms and collect a detailed inventory that can be further analyzed using the Migration Analytics application on cloud.redhat.com.
      </EmptyStateInfo>
      <EmptyStateAction
        className=""
        secondary={false}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsSize="large"
          bsStyle="primary"
          disabled={false}
          onClick={[Function]}
        >
          Get started
        </Button>
      </EmptyStateAction>
    </EmptyState>
    <h6
      className="manifest-version"
    >
      Manifest version: x.y.z 2019-06-10
    </h6>
  </AnalyticsContainer>
</React.Fragment>
`;
